python/pyspark/core/rdd.py

```diff
@@ -5370,6 +5370,17 @@ def _test() -> None:
     import tempfile
     from pyspark.core.context import SparkContext
 
+    try:
```

Comment:
shall we add a todo that once we upgrade the minimum version >= 2.0, we can remove this try-except and update the doc tests.

Reply:
There exist also multiple tests already with:
```python
try:
    # Numpy 1.14+ changed it's string format.
    numpy.set_printoptions(legacy="1.13")
except TypeError:
    pass
```
I'd guess these should be considered for updating before that ([since the minimum NumPy version is at `1.21` currently](https://github.com/apache/spark/blob/399980edaa81474b63fa6c7434fd47a1fb96f4d7/python/packaging/classic/setup.py#L154)).